### ISSUE TITLE
Ignore URL rewrites/redirects for non-WordPress URLs

### ISSUE THREAD

**Issue by dleigh** [oferwald/transposh#12]
Hi Ofer
On https://yuconvr.com, there are a set of "virtual reality" pages that are not in WordPress but are part of the domain.  For instance: 

https://yuconvr.com/VR/Piekstraat2

This gets redirected to (in English, for example): https://yuconvr.com/en/VR/Piekstraat2 and this will fail because that doesn't exist.  If I use the redirection functionalities of the Rank Math plugin, I can redirect https://yuconvr.com/en/VR/Piekstraat2 to https://yuconvr.com/VR/Piekstraat2 but it does a redirect loop back to https://yuconvr.com/en/VR/Piekstraat2. 

I ASSUME that it's Transposh still trying to redirect it to English when I've chosen the English flag. 

If that's so, is there a way to tell Transposh to leave it alone?
Thanks!

**Comment by dleigh** [comment]
So, I can see two solutions that I could implement.
1. Make sure that Transposh executes ahead of Rank Math's redirection code.  I was thinking of using this plugin to do that: https://wordpress.org/plugins/plugins-load-order/  In my thinking this would make sure that Tranposh doesn't try to redirect something that Rank Math has already redirected.  I imagine that there would be cases where I'd want the opposite.  Currently , however, there are no other redirections defined for this site.  I also don't know what other repercussions changing Transposh's execution order would have.   One advantage here is that I'm not altering plugin code.
2. Change this code in Transposh to ignore any "/VR/" URLs:
`    /**
     * Check if page is special (one that we normally should not touch
     * @param string $url Url to check
     * @return boolean Is it a special page?
     */
    function is_special_page($url) {
        return ( stripos($url, '/wp-login.php') !== FALSE ||
                stripos($url, '/robots.txt') !== FALSE ||
                stripos($url, '/wp-json/') !== FALSE ||
                stripos($url, '/wp-admin/') !== FALSE ||
                stripos($url, '/wp-comments-post') !== FALSE ||
                stripos($url, '/main-sitemap.xsl') !== FALSE || //YOAST?
                stripos($url, '.xsl') !== FALSE || //YOAST?
                stripos($url, '.xml') !== FALSE || //YOAST?
                stripos($url, '/xmlrpc.php') !== FALSE);
    }
`
Obviously, then each time Transposh updates, I'd have to re-add the code modifications.  Not ideal.

Any thoughts as to whether either of these ideas would be better or if there is another solution?

**Comment by dleigh** [comment]
Well, I tried both of these and neither worked, which surprised me - though Transposh is complex enough that this might not have been the right piece of code to change. 

At any rate, I found a good solution in the end.  The unwanted rewrite came from a link on an image created with the WP Bakery page builder.   The link was a full URL like:  https://yuconvr.com/en/VR/Piekstraat2.  I changed it to a relative URL like:  /VR/Piekstraat2 and it was not rewritten and worked perfectly.  Hopefully this helps someone else with this issue.
